### ISSUE TITLE
Column name error when saved to create new mysql table

### ISSUE THREAD

**Issue by tcak** [serge-rider/dbeaver#527]
I do not have any plan how to regenerate this issue, but it happened twice this morning.

I right click and create desired number of table rows. Dbeaver names them as "Column 1, Column 2, .." etc. Then I give their values in normal way. Create a primary key. Then I click on Save button.

Program creates the table, but gives an error (I didn't note it, if happens again, will post it here), saying something like column "Column 1 is not found.". On the table tab, "*" symbol is there. So, it indicates that Save didn't work. I click on OK on the coming dialog window, and close Dbeaver. When I reopen it, table is there and everything is created correctly.

When I click on Save button, there is no column with the name "Column 1" anywhere.

**Comment by carlosjrcabello** [comment]
Hi @tcak ,

Are you using spaces for columns names ? Did you get those errors in MySQL ?

Spaces are not recommended on SQL ANSI, but, you must use '`' when you need it:

`alter table `tb_foo` add column `my column` varchar(10);`

I believe DBeaver it's not closing column with '`';

**Comment by tcak** [comment]
No, there is no use of space in any column name at all. Only a-z0-9_  nothing else.

These days, there is no new project to create new tables. As soon as I get same error which happened many time before, I will put it here.

**Comment by serge-rider** [comment]
Well, I still can't reproduce that.
If you can - please reopen and provide steps to reproduce.

**Comment by tcak** [comment]
I have found the issue finally. I will explain what I have done, and list what SQL is generated. The problem comes from the generated SQL as far as I see.

I clicked on "Create table" on right click popup menu and entered the table name.

I quickly created 8 columns for the table and, their names were from "Column 1" to "Column 8".

I named them one by one. The "Column 7" was renamed as "sending_try_count" at first. After naming the "Column 8" as "sent_flag", I went back to "sending_try_count", and renamed it as "send_attempts". Then I created a primary key for "id", and two foreign keys.

At this point, I clicked on save button for the first time. A modal window came with the title "Persist changes". Below is the generated SQL code.

CREATE TABLE dbname.tbl9_email_crons (
	id int NOT NULL AUTO_INCREMENT,
	created_datetime datetime NOT NULL,
	sent_datetime datetime NULL,
	documents_id int NULL,
	emails_id int NULL,
	email varchar(128) NOT NULL,
	send_attempts tinyint unsigned NOT NULL,
	sent_flag enum('N','Y') NOT NULL,
	CONSTRAINT tbl9_email_crons_PK PRIMARY KEY (id),
	CONSTRAINT tbl9_email_crons_tbl4_company_documents_FK FOREIGN KEY (documents_id) REFERENCES dbname.tbl4_company_documents(id) ON DELETE RESTRICT ON UPDATE CASCADE,
	CONSTRAINT tbl9_email_crons_tbl6_emails_FK FOREIGN KEY (emails_id) REFERENCES dbname.tbl6_emails(id) ON DELETE RESTRICT ON UPDATE CASCADE
)
ENGINE=InnoDB
DEFAULT CHARSET=utf8
COLLATE=utf8_unicode_ci;
ALTER TABLE dbname.tbl9_email_crons CHANGE emails_oid emails_id int NULL;
ALTER TABLE dbname.tbl9_email_crons CHANGE Column6 email varchar(128) NOT NULL;
ALTER TABLE dbname.tbl9_email_crons CHANGE Column7 sending_try_count tinyint unsigned NOT NULL;
ALTER TABLE dbname.tbl9_email_crons CHANGE Column8 sent_flag enum('N','Y') NOT NULL;


The problem I was seeing is those extra ALTER TABLE expressions. They are not supposed to be there.
